eng/pipelines/common/xplat-setup.yml

```diff
@@ -171,9 +171,13 @@ jobs:
           demands: ImageOverride -equals Build.Ubuntu.2204.Amd64
 
         # OSX Build Pool (we don't have on-prem OSX BuildPool).
-        ${{ if in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator') }}:
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), eq(variables['System.TeamProject'], 'public')) }}:
           vmImage: 'macos-12'
 
+        # Official build OSX pool
+        ${{ if and(in(parameters.osGroup, 'osx', 'maccatalyst', 'ios', 'iossimulator', 'tvos', 'tvossimulator'), ne(variables['System.TeamProject'], 'public')) }}:
+          vmImage: 'macos-13-arm64'
```

Comment:
We should double that these builds use native Arm64 SDK after the switch.

(x64 SDK running on Arm64 emulator is slow and it has reliability issues.)